### PR TITLE
use modulekind to check if initializer for shorthand property assignm…

### LINKED ISSUE #5573: Shorthand property names issue with target ES6 and commonjs module
Using typescript@next (1.8.0-dev.20151109) and the following compilerOptions:
```json
"compilerOptions": {
    "target": "ES6",
    "module": "commonjs"
}
```
Any code using named import as a shorthand property will result in invalid code being generated.

Example:
```typescript
import {foo} from './foo';
const test = {
  foo
};
```
generates
```javascript
var foo_1 = require('./foo');
const test = {
    foo
};
```
```
ReferenceError: foo is not defined
```

### PR DESCRIPTION
**Pull request by vladima**
…ent should be emitted. Fixes #5573

### CODE REVIEW

File: tests/baselines/reference/shorthandPropertyAssignmentInES6Module.errors.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+tests/cases/compiler/test.ts(2,19): error TS2307: Cannot find module './foo2'.
+tests/cases/compiler/test.ts(6,1): error TS2304: Cannot find name 'console'.
+tests/cases/compiler/test.ts(7,1): error TS2304: Cannot find name 'console'.
```
Comment (sandersn) [inline tests/baselines/reference/shorthandPropertyAssignmentInES6Module.errors.txt]: Are these three errors part of the test? From reading the bug, it looks like only `foo1` and `const test = { x };` are needed and the rest can be removed.

**Comment by sandersn** [comment]
Looks good except for maybe some test cleanup.

File: tests/baselines/reference/shorthandPropertyAssignmentInES6Module.errors.txt (hunk 1 above)
Comment (vladima) [inline tests/baselines/reference/shorthandPropertyAssignmentInES6Module.errors.txt]: `console ` is irrelevant, However missing `./foo2` module is necessary to make sure that we emit the right thing no matter whether module is found or not.

Comment (sandersn) [inline tests/baselines/reference/shorthandPropertyAssignmentInES6Module.errors.txt]: Can you name the modules found.ts/notFound.ts then?

Comment (vladima) [inline tests/baselines/reference/shorthandPropertyAssignmentInES6Module.errors.txt]: sure

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**